### ISSUE TITLE
Only notify about dependencies that bumped to next major version

### ISSUE THREAD

**Issue by andreasabel** [haskell/hackage-server#1311]
I am getting notifications about bumping to `base-4.20` when the package says `base < 4.13` meaning it clearly is abandoned.

I'd like to get notifications only when my package has been up-to-date before.  So, e.g. when it says `base < 4.20` and then `base` is released with `4.20` or `5.x` or maybe even `4.21` in case `4.20` is deprecated or does not exist.

**Comment by gbaz** [comment]
That option should already exist? (Notification preferences are available from your user account page)

 
![image](https://github.com/haskell/hackage-server/assets/1125849/1d7a1075-f09b-4e7c-b713-4ae1ee37b76a)

**Comment by andreasabel** [comment]
Ah great, I am trying the third option then!

Now that I know about it, why did my former self raise the issue, and how can we prevent such in the future?

### Can the text of the notification be improved?

Currently it reads:
> The dependency [base-4.20.0.0](https://hackage.haskell.org/package/base-4.20.0.0) has been uploaded or revised.
>
> You have requested to be notified when a dependency isn't accepted by any of your maintained packages.
> 
> These are your packages that require base but don't accept 4.20.0.0:
>
>    [text-format-0.3.2.1](https://hackage.haskell.org/package/text-format-0.3.2.1)
>    [streaming-lzma-0.0.0.0](https://hackage.haskell.org/package/streaming-lzma-0.0.0.0)
>    [streaming-brotli-0.0.0.0](https://hackage.haskell.org/package/streaming-brotli-0.0.0.0)
>    [pipes-brotli-0.0.0.0](https://hackage.haskell.org/package/pipes-brotli-0.0.0.0)
>    [hgettext-0.1.40.1](https://hackage.haskell.org/package/hgettext-0.1.40.1)
>    [brotli-conduit-0.0.0.0](https://hackage.haskell.org/package/brotli-conduit-0.0.0.0)
>    [Agda-2.6.4.3](https://hackage.haskell.org/package/Agda-2.6.4.3)
>
> You can adjust your notification preferences at
> https://hackage.haskell.org/user/AndreasAbel/notify

It would be helpful if the list of packages could be divided into the two categories: 
1. _up-to-date_: They did accept the latest major version before the dependency was bumped.
2. _stale_: The already did not accept the latest major version.

(In this case, only 2 packages fall into the _up-to-date_ category and the other 5 are _stale_.)

The information about notification preferences could tell me that I can switch off notifications for _stale_ or for all packages.

### Can the notification settings page be improved?

> Notify when a dependency is updated for any of my maintained packages: 
> * Yes  
> * No
> trigger:
> * always
> * my maintained package doesn't accept the version of the newly uploaded or revised dependency
> * my maintained package doesn't accept the version of the newly uploaded or revised dependency and my package does accept the second highest version of said dependency

For one, it talks about "second highest version", so maybe it is not about _major_ versions.  But maybe I am mistaken about the major version idea, what counts is:
* it used to accept the latest version of the dependency, but
* it does not any more now.

I see some opportunities to reformulate the display of the options:
1. Make a single four stage escalation list from "always" down to "never", so, integrating the "No" option as "never".
2. Express the 3rd option in simpler terms, e.g.
    >  my maintained package doesn't accept the version of the newly uploaded or revised dependency but did just before the bump

The options would benefit from some help page with examples...
